ports/unix/main.c

```diff
@@ -642,6 +642,14 @@ MP_NOINLINE int main_(int argc, char **argv) {
         }
     }
 
+    #if MICROPY_PY_SYS_UATEXIT
+    if (mp_obj_is_callable(MP_STATE_VM(exitfunc))) {
+        mp_obj_t exitfunc = MP_STATE_VM(exitfunc);
+        MP_STATE_VM(exitfunc) = mp_const_none;
```

Comment:
This shouldn't be needed, it shouldn't matter that the exitfunc variable is still a callable, it won't be executed again.

Reply:
accepted